docgen/main.go

```diff
@@ -0,0 +1,153 @@
+// Generate markdown from the JSON schema for the Consul ECS config.
```

Comment:
in the k8s project i've been putting these things inside a `hack` folder. What do you think?

Reply:
Oh, fine with me. Maybe like `hack/generate-config-reference`?

Reply:
sounds good!